cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTableAdmin.java

```diff
@@ -56,6 +56,16 @@ public interface StructuredTableAdmin {
    */
   StructuredTableSchema getSchema(StructuredTableId tableId) throws IOException, TableNotFoundException;
 
+  /**
+   * Update a StructuredTable schema using the {@link StructuredTableSpecification}.
+   *
+   * @param spec table specification
+   * @throws IOException if there is an error for updating the table schema
+   * @throws TableNotFoundException if the table doesn't exist
+   */
+  void addColumnsIfNeeded(StructuredTableSpecification spec)
```

Comment:
simply call this `update`, and put in javadoc on what is allowed to update

Reply:
This method should be responsible to detect whether an update on columns is needed, spec should not handle it.

Reply:
got it